pkg/connector/login.go

```diff
@@ -5,43 +5,151 @@ import (
 	"fmt"
 
 	"maunium.net/go/mautrix/bridgev2"
+
+	"go.mau.fi/mautrix-meta/messagix"
+	"go.mau.fi/mautrix-meta/messagix/cookies"
+	"go.mau.fi/mautrix-meta/messagix/table"
+	"go.mau.fi/mautrix-meta/messagix/types"
 )
 
+var FACEBOOK_COOKIES_FLOW = "cookies-facebook"
+var INSTAGRAM_COOKIES_FLOW = "cookies-instagram"
+
 func (m *MetaConnector) CreateLogin(ctx context.Context, user *bridgev2.User, flowID string) (bridgev2.LoginProcess, error) {
-	if flowID != "cookies" {
+	if flowID != FACEBOOK_COOKIES_FLOW && flowID != INSTAGRAM_COOKIES_FLOW {
 		return nil, fmt.Errorf("unknown flow ID %s", flowID)
 	}
 
-	return &CookieLogin{}, nil
+	return &MetaCookieLogin{
+		Flow: flowID,
+		User: user,
+		Main: m,
+	}, nil
 }
 
 func (m *MetaConnector) GetLoginFlows() []bridgev2.LoginFlow {
-	return []bridgev2.LoginFlow{{
-		Name:        "Cookies",
-		Description: "Login using cookies from a web browser",
-		ID:          "cookies",
-	}}
+	facebook := bridgev2.LoginFlow{
+		Name:        "Facebook Cookies",
+		Description: "Login using cookies from Facebook Messenger",
+		ID:          FACEBOOK_COOKIES_FLOW,
+	}
+	instagram := bridgev2.LoginFlow{
+		Name:        "Instagram Cookies",
+		Description: "Login using cookies from Instagram",
+		ID:          INSTAGRAM_COOKIES_FLOW,
+	}
+	if m.Config == nil {
+		return []bridgev2.LoginFlow{facebook, instagram}
+	} else if m.Config.Mode == "facebook" {
+		return []bridgev2.LoginFlow{facebook}
+	} else if m.Config.Mode == "instagram" {
+		return []bridgev2.LoginFlow{instagram}
+	} else {
+		panic("unknown mode in config")
+	}
+}
+
+type MetaCookieLogin struct {
+	Flow string
+	User *bridgev2.User
+	Main *MetaConnector
 }
 
-type CookieLogin struct{}
+var _ bridgev2.LoginProcessCookies = (*MetaCookieLogin)(nil)
 
-var _ bridgev2.LoginProcessCookies = (*CookieLogin)(nil)
+func requiredCookiesToKeys(requiredCookies []cookies.MetaCookieName) []string {
+	var keys []string
+	// Turn the array of MetaCookieName into an array of strings
+	for _, key := range requiredCookies {
+		keys = append(keys, string(key))
+	}
+	return keys
+}
```

Comment:
I added a generic utility for this: `exslices.CastToString` should work with any string type aliases https://github.com/mautrix/go-util/commit/4fa40bf64652a2ac475e57b456cbfef4fb7ee1cf